openrr-apps/src/robot_config.rs

```diff
@@ -322,54 +666,149 @@ impl RobotConfig {
+            .filter(|c| {
+                self.joint_trajectory_clients
+                    .as_ref()
+                    .map_or(true, |v| v.contains(&c.name))
+            })
+            .cloned()
+            .collect();
+
+        let mut urdf_robot = None;
         #[cfg(not(feature = "ros"))]
-        let raw_joint_trajectory_clients = create_joint_trajectory_clients_lazy(
-            self.urdf_viz_clients_configs.clone(),
-            urdf_robot.as_ref(),
-        )?;
+        let use_urdf = !urdf_viz_clients_configs.is_empty();
         #[cfg(feature = "ros")]
-        let raw_joint_trajectory_clients = {
-            let mut clients = if self.urdf_viz_clients_configs.is_empty() {
-                HashMap::new()
-            } else {
-                create_joint_trajectory_clients_lazy(
-                    self.urdf_viz_clients_configs.clone(),
-                    urdf_robot.as_ref(),
-                )?
-            };
-            clients.extend(arci_ros::create_joint_trajectory_clients_lazy(
-                self.ros_clients_configs.clone(),
+        let use_urdf = !urdf_viz_clients_configs.is_empty() || !ros_clients_configs.is_empty();
+        if use_urdf {
+            if let Some(urdf_path) = self.openrr_clients_config.urdf_full_path() {
+                urdf_robot = Some(urdf_rs::utils::read_urdf_or_xacro(urdf_path)?);
+            }
+        }
+
+        let mut clients = if urdf_viz_clients_configs.is_empty() {
+            HashMap::new()
+        } else {
+            arci_urdf_viz::create_joint_trajectory_clients_lazy(
+                urdf_viz_clients_configs,
                 urdf_robot.as_ref(),
-            )?);
-            clients
+            )?
         };
-        Ok(raw_joint_trajectory_clients)
+
+        #[cfg(feature = "ros")]
+        clients.extend(arci_ros::create_joint_trajectory_clients_lazy(
+            ros_clients_configs,
+            urdf_robot.as_ref(),
+        )?);
+
+        for (plugin_name, config) in &self.plugins {
+            for instance in config.instances.iter().filter(|instance| {
+                instance.type_ == PluginInstanceKind::JointTrajectoryClient
+                    && self
+                        .joint_trajectory_clients
+                        .as_ref()
+                        .map_or(true, |v| v.contains(&instance.name))
+            }) {
+                if clients.contains_key(&instance.name) {
+                    return Err(Error::DuplicateInstance(format!(
+                        "Multiple {:?} instances {:?} are found. Consider renaming one of the instances",
+                        instance.type_, instance.name,
+                    )));
+                }
+                let plugin = plugins.load(plugin_name)?.unwrap();
+                let args = instance.load_args()?;
+                let plugin_name = plugin_name.clone();
+                let instance_name = instance.name.clone();
+                let instance_kind = instance.type_;
```

Comment:
This is repeated again and again. How about creating a function for this?

Reply:
done in https://github.com/openrr/openrr/pull/319/commits/4bb644a3649f6bf891b91438827b6336530b0a54